### DISCUSSION TITLE
A `--quiet` option?

### DISCUSSION THREAD

**Discussion by frederikhors** [SeaQL/sea-orm#1199] [Ideas]
I'm using cli with:

```shell
sea-orm-cli generate entity --date-time-crate time --database-url ... --output-dir ./entities
```

It prints A LOT (really, a lot) of info in my shell.

Is there a way to reduce to errors?

Something like `--quiet` of tools like `cargo fmt` would be awesome!

Thanks!

`sea-orm-cli 0.9.3`

**Comment by tyt2y3** [comment]
Oh sad, seems like cli is using println instead of log
________________________________
From: frederikhors ***@***.***>
Sent: Monday, October 17, 2022 7:48:47 AM
To: SeaQL/sea-orm ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [SeaQL/sea-orm] A `--quiet` option? (Issue #1126)


I'm using cli with:

sea-orm-cli generate entity --date-time-crate time --database-url ... --output-dir ./entities

It prints A LOT (really, a lot) of info in my shell.

is there a way to reduce to errors?

Something like --quiet of tools like cargo fmt would be awesome!

Thanks!

sea-orm-cli 0.9.3

**Comment by billy1624** [comment]
Hey @frederikhors, what do you mean by "errors"?

Do you mean:
1. The `Generating metadata.rs ...` part?
2. Or, the `Warning: can't set `format_code_in_doc_comments = true`, unstable features are only available in nightly channel.` part?

```
Generating metadata.rs
    > Column `uuid`: Uuid, not_null
    > Column `type`: String, not_null
    > Column `key`: String, not_null
    > Column `value`: String, not_null
    > Column `bytes`: Vec<u8>, not_null
    > Column `date`: Option<Date>
    > Column `time`: Option<Time>
Generating json_vec.rs
    > Column `id`: i32, auto_increment, not_null
    > Column `str_vec`: String, not_null
Warning: can't set `format_code_in_doc_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `format_code_in_doc_comments = true`, unstable features are only available in nightly channel.
```

**Comment by billy1624** [comment]
> Oh sad, seems like cli is using println instead of log

Nope, it's not.

https://github.com/SeaQL/sea-orm/blob/4cf58af4b1e9de61964aaa4de17aeb04c3ba28bb/sea-orm-codegen/src/entity/writer.rs#L143-L146

We use `tracing::info` to record the log. Then, print the message onto console with the writer. It's trivial to make it quiet, i.e. no printing.

https://github.com/SeaQL/sea-orm/blob/4cf58af4b1e9de61964aaa4de17aeb04c3ba28bb/sea-orm-cli/src/commands/generate.rs#L31-L47

**Comment by JonasCir** [comment]
Yes please, I wanted to ask for the same thing!

@billy1624 From how I know quiet modes, it would apply to both. Just nothing get's printed to stdout at all and people can only rely on the exit code to see it worked without  errors.

**Comment by frederikhors** [comment]
@billy1624 I said:

> reduce to errors

To reduce the noise and only print errors (if any).

**Comment by tyt2y3** [comment]
I would prefer using `RUST_LOG=error` to configure the logging.
Seems like sea-orm-migration uses println though, not sure if it's a deal breaker.

https://github.com/SeaQL/sea-orm/blob/712f3bf707f4512afc743971ad0f4d6ee851d542/sea-orm-cli/src/commands/migrate.rs#L68

**Reply by frederikhors** [reply]
Why did you close this and other issues?

**Reply by tyt2y3** [reply]
We will make sure logging is configuration via an env var, but I don't think a `--quiet` option is necessary.
And also make things print to stderr instead of stdout

**Reply by frederikhors** [reply]
> We will make sure logging is configuration via an env var

Is there any issue open to follow this?
